### PR TITLE
New Resource: `azurerm_service_fabric_mesh_local_network`

### PR DESCRIPTION
**Pull request by mbfrahry**
```
--- PASS: TestAccAzureRMServiceFabricMeshLocalNetwork_basic (88.18s)
--- PASS: TestAccAzureRMServiceFabricMeshLocalNetwork_update (122.01s)
```

### CODE REVIEW

**Review by jackofallops — APPROVED** [review]
Thanks @mbfrahry - Couple of questions / suggestions below, but otherwise LGTM.

Side note - Is it worth updating for the new file naming convention we've started moving to while it's a new resource? (i.e. `service_fabric_mesh_network_resource.go`)

File: azurerm/internal/services/servicefabricmesh/parse/service_fabric_mesh_network.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+	Name          string
+}
+
+func ServiceFabricMeshNetworkID(input string) (*ServiceFabricMeshNetworkId, error) {
+	id, err := azure.ParseAzureResourceID(input)
+	if err != nil {
+		return nil, fmt.Errorf("[ERROR] Unable to parse Service Fabric Mesh Network ID %q: %+v", input, err)
+	}
+
+	cluster := ServiceFabricMeshNetworkId{
```
[the remark this answers is not in the document]
Comment (mbfrahry) [reply azurerm/internal/services/servicefabricmesh/parse/service_fabric_mesh_network.go:20]: Good call. Updated!

File: azurerm/internal/services/servicefabricmesh/resource_arm_service_fabric_mesh_local_network.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+	if _, err := client.Create(ctx, resourceGroup, name, parameters); err != nil {
+		return fmt.Errorf("creating Service Fabric Mesh Local Network %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		return fmt.Errorf("retrieving Service Fabric Mesh Local Network %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	if resp.ID == nil {
```
[the remark this answers is not in the document]
Comment (mbfrahry) [reply azurerm/internal/services/servicefabricmesh/resource_arm_service_fabric_mesh_local_network.go:113]: Good call

File: azurerm/internal/services/servicefabricmesh/tests/resource_arm_service_fabric_mesh_local_network_test.go [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-sfm-%d"
+  location = "%s"
+}
+
+resource "azurerm_service_fabric_mesh_local_network" "test" {
+  name                   = "accTest-%d"
+  resource_group_name    = azurerm_resource_group.test.name
+  location               = azurerm_resource_group.test.location
+  network_address_prefix = "10.0.0.0/22"
```
[the remark this answers is not in the document]
Comment (mbfrahry) [reply azurerm/internal/services/servicefabricmesh/tests/resource_arm_service_fabric_mesh_local_network_test.go:161]: Checked and it works!

**Pull request merged**